src/wazuh_modules/vulnerability_detector/wm_vuln_detector_nvd.c

```diff
@@ -1988,6 +2027,152 @@ int wm_vuldet_report_nvd_vulnerabilities(sqlite3 *db, vu_nvd_report **nvd_report
     return retval ? wm_vuldet_sql_error(db, stmt) : 0;
 }
 
+char *wm_vuldet_clean_version(const char *version) {
+    char *data = NULL;
+    char *dup = NULL;
+
+    os_strdup(version, dup);
+
+    // Store the data between : and -
+    strtok(dup, ":");
+    data = strtok(NULL, ":");
+    data = strtok((data) ? data : dup, "-");
+
+    os_free(dup);
+    return data;
+}
+
+bool wm_vuldet_check_generic_package(sqlite3_stmt *pStmt, sqlite3 *dbCVE, OSHash *cve_table, int8_t name_type) {
+    bool vulnerable_global = false;
+    bool vulnerable = false;
+    cve_vuln_pkg *newPkg = NULL;
+    char *pkg_reference = NULL;
+    char *pkg_source = NULL;
+    char *pkg_name = NULL;
+    char *pkg_version = NULL;
+
+    sqlite_strdup((char *)sqlite3_column_text(pStmt, PACKAGE_SOURCE), pkg_source);
+    sqlite_strdup((char *)sqlite3_column_text(pStmt, PACKAGE_NAME), pkg_name);
+    sqlite_strdup((char *)sqlite3_column_text(pStmt, PACKAGE_VERSION), pkg_version);
+
+    pkg_reference = (name_type == PACKAGE_SOURCE) ? pkg_source : pkg_name;
+
+    int res = sqlite3_prepare_v2(dbCVE, vu_queries[VU_GET_GENERIC_PACKAGE], -1, &pStmt, NULL);
+    if (res != SQLITE_OK) {
+        return wm_vuldet_sql_error(dbCVE, pStmt);
+    } else {
+        sqlite3_bind_text(pStmt, 1, pkg_reference, -1, NULL);
+        sqlite3_bind_text(pStmt, 2, "*", -1, NULL);
```

Comment:
Why don't you put this asterisk in the query definition? Does it need to be a parameter?